src/core/server/http/http_service.ts

```diff
@@ -101,11 +142,18 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
     const contract: HttpServiceSetup = {
       ...serverContract,
 
-      createRouter: (path: string) => {
-        const router = new Router(path, this.log);
+      createRouter: (path: string, pluginId: PluginOpaqueId = this.coreContext.coreId) => {
+        const enhanceHandler = this.requestHandlerContext!.createHandler.bind(null, pluginId);
```

Comment:
I like the `enhance` name, maybe we should consider renaming `IContextContainer.createHandler` to `IContextContainer.enhanceHandler`